Repo: flat-config-types integration tests failing on main: Type 'FlatConfig' is not generic

### Suggestion

https://github.com/typescript-eslint/typescript-eslint/actions/runs/10775346911/job/29879715875 -> https://cloud.nx.app/runs/wStZYo8iWI:

Three tests failing, such as:

```plaintext
  ● flat-config-types-@types__eslint-v8 › typescript with --strictNullChecks, --exactOptionalPropertyTypes › should work successfully

    expect(received).toHaveLength(expected)

    Expected length: 0
    Received length: 5
    Received array:  ["node_modules/@stylistic/eslint-plugin/dist/dts/index.d.ts(86,79): error TS2315: Type 'FlatConfig' is not generic.", "node_modules/@stylistic/eslint-plugin/dist/dts/index.d.ts(86,119): error TS2315: Type 'FlatConfig' is not generic.", "node_modules/@stylistic/eslint-plugin/dist/dts/index.d.ts(92,23): error TS2315: Type 'FlatConfig' is not generic.", "node_modules/@stylistic/eslint-plugin/dist/dts/index.d.ts(100,25): error TS2315: Type 'FlatConfig' is not generic.", "node_modules/@stylistic/eslint-plugin/dist/dts/index.d.ts(108,17): error TS2315: Type 'FlatConfig' is not generic."]
```

### Additional Info

Relevant to #8603 for either filing  an issue when integration tests fail on main, or pinning and auto-updating dependencies.

💖

--- a/packages/integration-tests/fixtures/flat-config-types-@types__eslint-v8/package.json
+++ b/packages/integration-tests/fixtures/flat-config-types-@types__eslint-v8/package.json
@@ -6,8 +6,8 @@
     "@types/eslint__js": "latest",
     "@eslint/js": "latest",
     "@types/eslint": "^8",
-    "eslint": "latest",
-    "@stylistic/eslint-plugin": "latest",
+    "eslint": "9.9.1",
+    "@stylistic/eslint-plugin": "2.3.0",
     "eslint-plugin-deprecation": "latest",
     "eslint-plugin-jest": "latest"
   }

--- a/packages/integration-tests/fixtures/flat-config-types-@types__eslint-v8/package.json
+++ b/packages/integration-tests/fixtures/flat-config-types-@types__eslint-v8/package.json
@@ -7,7 +7,7 @@
     "@eslint/js": "latest",
     "@types/eslint": "^8",
     "eslint": "latest",
-    "@stylistic/eslint-plugin": "2.3.0",
+    "@stylistic/eslint-plugin": "latest",
     "eslint-plugin-deprecation": "latest",
     "eslint-plugin-jest": "latest"
   }

